src/lib.rs

```diff
@@ -70,11 +74,18 @@ impl log::Log for FilePerThreadLogger {
     fn log(&self, record: &Record) {
         if self.enabled(record.metadata()) {
             WRITER.with(|rc| {
+                if rc.borrow().is_none() {
+                    rc.replace(Some(open_file("")));
+                }
```

Comment:
I don't think this is quite right: it's explicitly said (in `initialize`'s doc comment, as well as in the panic below) that the `initialize` function must be called before logging. Can you revert this chunk, please?

Reply:
Ah yes. I made this change later a bit later on. It was because of this: One of the dependencies creates separate threads to perform certain operations. I'm interested in the logs written from those threads as well.

Do you think there's a better way to handle this (without the panic) ?

Reply:
Ah, that's an interesting use case. It's a bit unfortunate that files pop up without any explanation, especially since this crate's API contract is to explicitly call `initialize`. We might need something more elaborate for covering this use case, though: a static atomic boolean that controls whether it's allowed to not initialize. Then the process can call this new API function before setting any logging and spawning threads, setting this atomic value to true. Threads can then read the value here, with something resembling this:

```
if rc.borrow().is_none() && ALLOW_UNINITIALIZED {
  // your code
}
```

Does it make sense? I'd rather keep the current behavior for our use case, we don't want to accidentally spam users hard drives with log files :-)

Reply:
Yes that's understandable. I'll give that a shot in a separate PR. :)